lib/playdate_web/components/nickname_form.ex

```diff
@@ -0,0 +1,48 @@
+defmodule PlaydateWeb.Components.NicknameForm do
+  @moduledoc """
+  Displays the nickname choosing form for the app lobby.
+  """
+  use PlaydateWeb, :component
+  alias Phoenix.LiveView.{Rendered, Socket}
+  alias Surface.Components.Form
+
+  prop form, :struct
+  prop app_name, :string
+
+  @doc false
+  @impl true
+  @spec render(Socket.assigns()) :: Rendered.t()
+  def render(assigns) do
+    ~F"""
+    <Form for={@form} submit="update-nickname">
```

Comment:
```suggestion
    <Form for={@form} submit={@submit}>
```

To make the contract explicit between the LiveView/Component that hosts this form and the submit event it will generate.